zerver/views/webhooks/librato.py

```diff
@@ -0,0 +1,183 @@
+        return content
+
+    def handle_snapshots(self):
+        # type: () -> text_type
+        content = u''
+        for attachment in self.attachments:
+            content += self.handle_snapshot(attachment)
+        return content
+
+    def handle_snapshot(self, snapshot):
+        # type: (Dict[str, Any]) -> text_type
+        snapshot_template = u"**{author_name}** send a [snapshot]({image_url}) of [metric]({title})"
+        author_name, image_url, title = self.parse_snapshot(snapshot)
+        content = snapshot_template.format(author_name=author_name, image_url=image_url, title=title)
+        return content
+
+    def handle_alert_violation_message(self):
+        # type: () -> text_type
+        alert_violation_template = u"Alert [alert_name]({alert_url}) has triggered! "
+        alert_id, alert_name, alert_url, alert_runbook_url = self.parse_alert()
+        content = alert_violation_template.format(alert_name=alert_name, alert_url=alert_url)
+        if alert_runbook_url:
+            alert_runbook_template = u"[Reaction steps]({alert_runbook_url})"
+            content += alert_runbook_template.format(alert_runbook_url=alert_runbook_url)
+        content += self.generate_conditions_and_violations()
+        return content
+
+    def generate_conditions_and_violations(self):
+        # type: () -> text_type
+        conditions = self.parse_conditions()
+        violations = self.parse_violations()
+        content = u""
+        for condition, violation in zip(conditions, violations):
+            content += self.generate_violated_metric_condition(violation, condition)
+        return content
+
+    def generate_violated_metric_condition(self, violation, condition):
+        # type: (Dict[str, Any], Dict[str, Any]) -> text_type
+        summary_function, threshold, condition_type, duration = self.parse_condition(condition)
+        metric_name, recorded_at = self.parse_violation(violation)
+        metric_condition_template = u"\n>Metric `{metric_name}`, {summary_function} was {condition_type} {threshold}"
+        content = metric_condition_template.format(
+                metric_name=metric_name, summary_function=summary_function, condition_type=condition_type,
+                threshold=threshold)
+        if duration:
+            content += u" by {duration}s".format(duration=duration)
+        content += u", recorded at {recorded_at}".format(recorded_at=recorded_at)
+        return content
+
+@api_key_only_webhook_view('Librato')
+@has_request_variables
+def api_librato_webhook(request, user_profile, client, payload=REQ(converter=ujson.loads, default={}),
+                        stream=REQ(default='librato'), topic=REQ(default=None)):
+    # type: (HttpRequest, UserProfile, Client, Dict[str, Any], text_type, text_type) -> HttpResponse
+    try:
+        attachments = ujson.loads(request.body).get('attachments', [])
+    except ValueError:
+        attachments = []
```

Comment:
I think this `request.body` stuff can just use `REQ(argument_type=body)`, right?